### ISSUE TITLE
[Cisco FTD] parsing issue with Logstash

### ISSUE THREAD

**Issue by daixque** [elastic/integrations#7159]
# Overview

Pipeline error happens when we use Cisco FTD integration with Logstash. Data flow looks like below.

Device -> Elastic Agent (Cisco FTD integration) -> Logstash (no special processing/filters) -> Elasticsearch (ESS)

On this configuration, we faced below error.

<img width="725" alt="image" src="https://github.com/elastic/integrations/assets/622436/ff698d21-2422-47e4-b347-ff3bcb004932">

# RCA

So far analysis is not enough, but we found Logstash will add event.original field automatically, and ingest pipeline of the integration expects no event.original field exists on original message.

I tried to add remove processor in the beginning of the pipeline to drop event.original field, but then following parsing might not going well because cisco.ftd object is gone after the change.

Example of their document which is stored on ES is:

```
{
  "_index": ".ds-logs-cisco_ftd.log-default-2023.07.26-000006",
  "_id": "aFXnlYkBB_z4EpW3BPIj",
  "_version": 1,
  "_score": 0,
  "_source": {
    "agent": {
      "name": "xxx",
      "id": "aa0250f4-86a9-44b8-a0bb-12c8c01f8d1f",
      "ephemeral_id": "2e818508-de2b-4ff5-9fb7-da965bd7b3cd",
      "type": "filebeat",
      "version": "8.8.2"
    },
    "log": {
      "source": {
        "address": "123.247.125.43:62657"
      }
    },
    "elastic_agent": {
      "id": "aa0250f4-86a9-44b8-a0bb-12c8c01f8d1f",
      "version": "8.8.2",
      "snapshot": false
    },
    "message": "<190>Jul 27 2023 09:36:53 Firepower-module1 : %FTD-6-302024: Built director stub UDP connection for inside:123.245.65.52/55936 (123.245.65.52/55936) to inside:123.245.152.16/53 (123.245.152.16/53)",
    "error": {
      "message": [
        "field [event.original] already exists"
      ]
    },
    "tags": [
      "private_is_internal",
      "cisco-ftd",
      "forwarded",
      "beats_input_codec_plain_applied"
    ],
    "input": {
      "type": "tcp"
    },
    "@timestamp": "2023-07-27T05:50:40.932Z",
    "ecs": {
      "version": "8.0.0"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "cisco_ftd.log"
    },
    "@version": "1",
    "host": {
      "address": "123.247.125.43"
    },
    "event": {
      "agent_id_status": "auth_metadata_missing",
      "ingested": "2023-07-27T05:51:32Z",
      "original": "<190>Jul 27 2023 09:36:53 Firepower-module1 : %FTD-6-302024: Built director stub UDP connection for inside:123.245.65.52/55936 (123.245.65.52/55936) to inside:123.245.152.16/53 (123.245.152.16/53)",
      "timezone": "+08:00",
      "kind": "pipeline_error",
      "dataset": "cisco_ftd.log"
    }
  },
  "fields": {
    "elastic_agent.version": [
      "8.8.2"
    ],
    "host.address": [
      "123.247.125.43"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "cisco_ftd"
    ],
    "@version": [
      "1"
    ],
    "agent.name": [
      "xxx"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "event.agent_id_status": [
      "auth_metadata_missing"
    ],
    "event.kind": [
      "pipeline_error"
    ],
    "event.timezone": [
      "+08:00"
    ],
    "event.original": [
      "<190>Jul 27 2023 09:36:53 Firepower-module1 : %FTD-6-302024: Built director stub UDP connection for inside:123.245.65.52/55936 (123.245.65.52/55936) to inside:123.245.152.16/53 (123.245.152.16/53)"
    ],
    "elastic_agent.id": [
      "aa0250f4-86a9-44b8-a0bb-12c8c01f8d1f"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "input.type": [
      "tcp"
    ],
    "message": [
      "<190>Jul 27 2023 09:36:53 Firepower-module1 : %FTD-6-302024: Built director stub UDP connection for inside:123.245.65.52/55936 (123.245.65.52/55936) to inside:123.245.152.16/53 (123.245.152.16/53)"
    ],
    "data_stream.type": [
      "logs"
    ],
    "tags": [
      "private_is_internal",
      "cisco-ftd",
      "forwarded",
      "beats_input_codec_plain_applied"
    ],
    "event.ingested": [
      "2023-07-27T05:51:32Z"
    ],
    "@timestamp": [
      "2023-07-27T05:50:40.932Z"
    ],
    "agent.id": [
      "aa0250f4-86a9-44b8-a0bb-12c8c01f8d1f"
    ],
    "ecs.version": [
      "8.0.0"
    ],
    "log.source.address": [
      "123.247.125.43:62657"
    ],
    "error.message": [
      "field [event.original] already exists"
    ],
    "data_stream.dataset": [
      "cisco_ftd.log"
    ],
    "agent.ephemeral_id": [
      "2e818508-de2b-4ff5-9fb7-da965bd7b3cd"
    ],
    "agent.version": [
      "8.8.2"
    ],
    "event.dataset": [
      "cisco_ftd.log"
    ]
  }
}
```

**Comment by andrewkroh** [comment]
Relates to:

- https://github.com/elastic/integrations/issues/3451#issuecomment-1163975845
- https://github.com/logstash-plugins/logstash-input-beats/issues/465

IIUC, to stop Logstash from mutating the event coming from Agent set [enrich](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-beats.html#plugins-inputs-beats-enrich) to `none`.

```
input {
  beats {
    port => 1234
    enrich => none
  }
}
```

**Comment by andrewkroh** [comment]
Try that and see let us know if it fixes the issue.

**Comment by daixque** [comment]
Just update, once enrich => none option is given, no messages will come to Elasticsearch's index. Not sure why so far. (It may drop at ingest pipeline..?)

**Comment by elangoshifts** [comment]
I am also facing the exact same issue, the weird thing here is when i edit the default pipeline and mark the rename as "ignore_error", the input fails in logstash.

I see logstash facing 400 errors saying that the mapping is incorrect, this happens because multiple filebeat lines are being merged together.

my flow is Cisco FTD -> Syslog -> Beats (tcp) -> LS -> ES

Scenario 1:
All defaults, enrich is enabled in LS

[field already exists errors] in ES (Kibana) => Pipeline failure


Scenario 2:
Update the default pipeline and added a remove processor to remove "event.original" as the first processor.

Following this the events now start failing when logstash send them to ES. 

Could not index event to Elasticsearch. {:status=>400, :action=>["create"......

It fails because, i can see two FTD lines concatenated together like below

<line 1 partial line cut ><190><line 2>

example

ResponderPackets: 1, InitiatorBytes: 128, **RespondSep 04** 2023 02:06:54 Firepower-module1  %FTD-6-430002: DeviceUUID: d4c57a3a-fa7e-11ea-


the <190> is present in some cases between the two lines

**Comment by elangoshifts** [comment]
Update on scenario 2. I initially thought that all events are being merged and rejected by ES, later identified that only some events are merged and being rejected. The reason i didn't notice the successfully processed events was the timezone parsing was incorrect and hence the events showed up in a different time range.

The only open issue now is that certain events are being merged with like below
<event 1 partial line cut ><190><event line 2>

**Comment by taylor-swanson** [comment]
This should be fixed by #8269 (available in cisco_ftd `2.19.0` or later).
